src/Raven.Server/Documents/Handlers/Debugging/NodeDebugHandler.cs

```diff
@@ -197,44 +239,49 @@ await using (var writer = new AsyncBlittableJsonTextWriter(ctx, stream))
 
                                     case TcpConnectionStatus.AuthorizationFailed:
-                                        result.Error = $"Connection to {url} failed because of authorization failure: {response.Message}";
-                                        logs?.Add(result.Error);
-                                        throw new AuthorizationException(result.Error);
+                                        message = $"Connection to {url} failed because of authorization failure: {response.Message}";
+                                        errors.Add(message);
+                                        logs?.Add(message);
+                                        throw new AuthorizationException(message);
 
                                     case TcpConnectionStatus.TcpVersionMismatch:
-                                        result.Error = $"Connection to {url} failed because of mismatching tcp version: {response.Message}";
-                                        logs?.Add(result.Error);
-                                        throw new AuthorizationException(result.Error);
+                                        message = $"Connection to {url} failed because of mismatching tcp version: {response.Message}";
+                                        errors.Add(message);
+                                        logs?.Add(message);
+                                        throw new AuthorizationException(message);
 
                                     case TcpConnectionStatus.InvalidNetworkTopology:
-                                        result.Error = $"Connection to {url} failed because of {nameof(TcpConnectionStatus.InvalidNetworkTopology)} error: {response.Message}";
-                                        logs?.Add(result.Error);
-                                        throw new InvalidNetworkTopologyException(result.Error);
-
+                                        message = $"Connection to {url} failed because of {nameof(TcpConnectionStatus.InvalidNetworkTopology)} error: {response.Message}";
+                                        errors.Add(message);
+                                        logs?.Add(message);
+                                        throw new InvalidNetworkTopologyException(message);
                                 }
                             }
-                        }
-                        catch (Exception e)
-                        {
-                            result.Error = e.ToString();
-                            logs?.Add($"Error occurred while attempting to negotiate with the server. {e.Message}");
-                            throw;
-                        }
 
-                        return null;
+                            return null;
+                        }
                     }
                 }
             }
+            catch (Exception e)
+            {
+                result.Errors = errors;
+                result.Errors.Add(e.ToString());
```

Comment:
Let's save the error here if it wasn't already set.

Reply:
done